appinventor/blocklyeditor/src/field_flydown.js

```diff
@@ -34,11 +35,11 @@ goog.require('Blockly.Options');
  * @extends {Blockly.FieldTextInput}
  * @constructor
  */
-Blockly.FieldFlydown = function(name, isEditable, displayLocation, opt_changeHandler) {
+Blockly.FieldFlydown = function(name, isEditable, opt_displayLocation, opt_changeHandler) {
   // This by itself does not control editability
   this.EDITABLE = isEditable;
   // [lyn, 10/27/13] Make flydown direction an instance variable
-  this.displayLocation = displayLocation;
+  this.displayLocation = opt_displayLocation;
```

Comment:
It's not necessary in this case because [the logic](https://github.com/mit-cml/appinventor-sources/blob/c796aca85faa57f69ba214767a20636f9c0a3645/appinventor/blocklyeditor/src/field_flydown.js#L176) uses and else for the default.
```
  if (this.displayLocation === Blockly.FieldFlydown.DISPLAY_BELOW) {
    y = y + borderBBox.height * flydown.workspace_.scale;
  } else { // if (this.displayLocation === Blockly.FieldFlydown.DISPLAY_RIGHT) {
    x = x + borderBBox.width * flydown.workspace_.scale;
  }
```

We could make it explicit if you want though.

Reply:
Understanding the default behavior via the logic you highlighted is subtle and nonobvious. Please instead use the conditional expression I recommended.

Reply:
[Done =)](https://github.com/mit-cml/appinventor-sources/pull/2215/commits/2ad8a205d6ab74ac7327de118baa9f78ecf897a6)